### PR TITLE
Respect Spark bundled log4j in extension modules

### PR DESCRIPTION
**Pull request by pan3793**
### _Why are the changes needed?_

As Spark 3.1/3.2 still use log4j 1.2, we should respect Spark bundled log4j in extension modules.

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: .github/workflows/master.yml [hunk 1]
```diff
@@ -73,7 +73,6 @@ jobs:
           -Drat.skip=true
           -Dscalastyle.skip=true
           -Dspotless.check.skip
-          -Dorg.slf4j.simpleLogger.defaultLogLevel=warn
```
Comment (yaooqinn) [inline .github/workflows/master.yml:76]: why do we bring  the maven build info log back

Comment (pan3793) [reply .github/workflows/master.yml:76]: To make the log clearly show failed in which module

Comment (yaooqinn) [reply .github/workflows/master.yml:76]: are the test classname and test name insufficient ?

Comment (pan3793) [reply .github/workflows/master.yml:76]: sufficient but not clear

Comment (yaooqinn) [reply .github/workflows/master.yml:76]: Why are the test classname and test name sufficient but not clear? which part is unclear? can you explain more, it is quite hard to follow if the comment is too short.

Comment (yaooqinn) [reply .github/workflows/master.yml:76]: > To make the log clearly show failure occurs in which module


You actually clearly get the module name `kyuubi-server_2.12`, class name and test name very easily.


```scala
AllKyuubiConfiguration:
- Check all kyuubi configs *** FAILED ***
  java.lang.AssertionError: assertion failed: /home/runner/work/incubator-kyuubi/incubator-kyuubi/docs/deployment/settings.md out of date, please update doc with KYUUBI_UPDATE=1 build/mvn clean install -Pflink-provided,spark-provided -DwildcardSuites=org.apache.kyuubi.config.AllKyuubiConfiguration
  at scala.Predef$.assert(Predef.scala:223)
  at org.apache.kyuubi.TestUtils$.verifyOutput(TestUtils.scala:57)
  at org.apache.kyuubi.config.AllKyuubiConfiguration.$anonfun$new$1(AllKyuubiConfiguration.scala:264)
  at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
  at org.scalatest.OutcomeOf.outcomeOf(OutcomeOf.scala:85)
  at org.scalatest.OutcomeOf.outcomeOf$(OutcomeOf.scala:83)
  at org.scalatest.OutcomeOf$.outcomeOf(OutcomeOf.scala:104)
  at org.scalatest.Transformer.apply(Transformer.scala:22)
  at org.scalatest.Transformer.apply(Transformer.scala:20)
  at org.scalatest.funsuite.AnyFunSuiteLike$$anon$1.apply(AnyFunSuiteLike.scala:226)
  ...
KyuubiOperationEnginePoolSuite:
- ensure app name contains engine-pool when engine pool is enabled.
- ensure the sub-domain doesn't work with the CONNECTION share level.
HudiOperationSuite:
Run completed in 14 minutes, 38 seconds.
Total number of tests run: 277
Suites: completed 44, aborted 0
Tests: succeeded 274, failed 3, canceled 0, ignored 2, pending 0
*** 3 TESTS FAILED ***
Error:  Failed to execute goal org.scalatest:scalatest-maven-plugin:2.0.2:test (test) on project kyuubi-server_2.12: There are test failures -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.scalatest:scalatest-maven-plugin:2.0.2:test (test) on project kyuubi-server_2.12: There are test failures
```

Comment (pan3793) [reply .github/workflows/master.yml:76]: I can not remember which module the suite belong to for all classes, for `InsertShuffleNodeBeforeJoinSuite`, I can not know it belongs to `kyuubi-extension-spark-common` or `kyuubi-extension-spark-3-1` before searching codebase. And I don't think the maven info level log is too noisy, but gives much familiar information as run test in local in normal.

Comment (yaooqinn) [reply .github/workflows/master.yml:76]: https://github.com/apache/spark/blob/master/.github/workflows/build_and_test.yml#L624 FYI

Comment (pan3793) [reply .github/workflows/master.yml:76]: Your concern is reasonable, personally I prefer to info level, I keep my opinion and will revert this change to make it as is.

**Review by ulysses-you — APPROVED** [review]

**Comment by yaooqinn** [comment]
thanks, merged to master

**Comment by pan3793** [comment]
Cherry picked this patch to branch-1.5
